### ISSUE TITLE
[Fixed Form Parser] Error when '\n' does not exist at the end of program

### ISSUE THREAD

**Issue by KGB99** [lfortran/lfortran#4393]
I have created a small sample program which recreates the error:
```
      PROGRAM EXAMPLE_ERROR

            CHARACTER(2) X
            X = 'Hi'
            PRINT *, X

      END PROGRAM EXAMPLE_ERROR
```

I believe this is semantically the same to:
```
      PROGRAM EXAMPLE_ERROR

            CHARACTER(len=2) X
            X = 'Hi'
            PRINT *, X

      END PROGRAM EXAMPLE_ERROR
```

When compiling with `gfortran` the output file reads `Hi` as expected. When compiling with `lfortran --fixed-form` the error message is the following:
```
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/kerimbirgi/dev/lfortran/src/bin/lfortran.cpp", line 2478
    LCompilers::print_stack_on_segfault();
  File "/Users/kerimbirgi/dev/lfortran/src/bin/lfortran.cpp", line 2442
    err = compile_to_object_file(arg_file, tmp_o, false,
  File "/Users/kerimbirgi/dev/lfortran/src/bin/lfortran.cpp", line 912
    result = fe.get_asr2(input, lm, diagnostics);
  File "/Users/kerimbirgi/dev/lfortran/src/lfortran/fortran_evaluator.cpp", line 246
    res = get_ast2(code_orig, lm, diagnostics);
  File "/Users/kerimbirgi/dev/lfortran/src/lfortran/fortran_evaluator.cpp", line 213
    res = LFortran::parse(al, *code, diagnostics, compiler_options);
  File "/Users/kerimbirgi/dev/lfortran/src/lfortran/parser/parser.cpp", line 124
    if (!p.parse(s)) {
  File "/Users/kerimbirgi/dev/lfortran/src/lfortran/parser/parser.cpp", line 173
    if (!f_tokenizer.tokenize_input(diag, m_a)) return false;
  File "/Users/kerimbirgi/dev/lfortran/src/lfortran/parser/fixedform_tokenizer.cpp", line 1577
    f.t.cur_line = string_start;
  File "/Users/kerimbirgi/dev/lfortran/src/lfortran/parser/fixedform_tokenizer.cpp", line 1560
    next_line(next);
  File "/Users/kerimbirgi/dev/lfortran/src/lfortran/parser/fixedform_tokenizer.cpp", line 1538
    lex_program(cur, true);
  File "/Users/kerimbirgi/dev/lfortran/src/lfortran/parser/fixedform_tokenizer.cpp", line 1402
    push_token_advance(cur, "endprogram");
  File "/Users/kerimbirgi/dev/lfortran/src/lfortran/parser/fixedform_tokenizer.cpp", line 703
    LCOMPILERS_ASSERT(*(t.cur-1) == '\n')
AssertFailed: *(t.cur-1) == '\n'
```
The error seems to come from the fixedform_tokenizer.

**Comment by kmr-srbh** [comment]
#3738 seems to be the same issue in release mode.

**Comment by gxyd** [comment]
We can probably close the issue, as it has been fixed by #4432

**Comment by kmr-srbh** [comment]
Closing with reference to #4432.
